spark/src/main/java/org/apache/iceberg/spark/SparkWriteOptions.java

```diff
@@ -47,4 +47,7 @@ private SparkWriteOptions() {
 
   // Checks if input schema and table schema are same(default: true)
   public static final String CHECK_ORDERING = "check-ordering";
+
+  // File scan task set ID that is being rewritten
+  public static final String REWRITTEN_FILE_SCAN_TASK_SET_ID = "rewritten-file-scan-task-set-id";
```

Comment:
Should we be sharing this property key with the read? Maybe it should be belong to the file-scan-task object itself?

Reply:
I am open to alternatives. It is a write option so I've added it here to match the read option.

Reply:
Could you comment it here why the file scan task set id is required? Maybe I don't have enough context, but it is a bit hard to understand the purpose of this option.

Reply:
Added a bit of context. The right place for the config is still to be discussed.

Reply:
Thoughts on the name and where to put it, @rdblue?